pandas/_libs/tslibs/offsets.pyx

```diff
@@ -618,10 +606,11 @@ cdef class BaseOffset:
             When the specific offset subclass does not have a vectorized
             implementation.
         """
-        raise NotImplementedError(  # pragma: no cover
-            f"DateOffset subclass {type(self).__name__} "
-            "does not have a vectorized implementation"
-        )
+        warnings.warn("'Offset.apply_index(other)' is deprecated. "
```

Comment:
Probably need a whatsnew note about this since it appears `DateOffset.apply_index` is publicly documented:https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.tseries.offsets.DateOffset.apply_index.html

Reply:
the warning was previously issued in the decorator which this PR removed